core/test/solver/batch_lower_trs.cpp

```diff
@@ -0,0 +1,233 @@
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include <ginkgo/core/solver/batch_lower_trs.hpp>
+
+
+#include <gtest/gtest.h>
+
+
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/matrix/batch_dense.hpp>
+#include <ginkgo/core/matrix/batch_diagonal.hpp>
+#include <ginkgo/core/preconditioner/batch_jacobi.hpp>
+
+
+#include "core/test/utils.hpp"
+#include "core/test/utils/batch.hpp"
+
+
+namespace {
+
+
+template <typename T>
+class BatchLowerTrs : public ::testing::Test {
+protected:
+    using value_type = T;
+    using real_type = gko::remove_complex<T>;
+    using Mtx = gko::matrix::BatchCsr<value_type>;
+    using ubatched_mat_type = typename Mtx::unbatch_type;
+    using Dense = gko::matrix::BatchDense<value_type>;
+    using Solver = gko::solver::BatchLowerTrs<value_type>;
+
+    BatchLowerTrs()
+        : exec(gko::ReferenceExecutor::create()),
+          mtx(get_csr_lower_matrix()),
+          batchlowertrs_factory(Solver::build().on(exec)),
+          solver(batchlowertrs_factory->generate(mtx))
+    {}
+
+    std::shared_ptr<const gko::Executor> exec;
+    const gko::size_type nbatch = 2;
+    const int nrows = 4;
+    std::shared_ptr<Mtx> mtx;
+    std::unique_ptr<typename Solver::Factory> batchlowertrs_factory;
+    std::unique_ptr<gko::BatchLinOp> solver;
+
+    std::unique_ptr<Mtx> get_csr_lower_matrix()
+    {
+        auto mat = Mtx::create(gko::as<const gko::ReferenceExecutor>(exec), 2,
+                               gko::dim<2>(4, 4), 7);
+        int* const row_ptrs = mat->get_row_ptrs();
+        int* const col_idxs = mat->get_col_idxs();
+        value_type* const vals = mat->get_values();
+        // clang-format off
+        row_ptrs[0] = 0;row_ptrs[1] = 1; row_ptrs[2] = 3;
+        row_ptrs[3] = 5; row_ptrs[4] = 7;
+        col_idxs[0] = 0; col_idxs[1] = 0; col_idxs[2] = 1;
+        col_idxs[3] = 0;  col_idxs[4] = 2; col_idxs[5] = 2; col_idxs[6] = 3;
+        vals[0] = 2.0; vals[1] = 1.0; vals[2] = 2.0; vals[3] = 4.0;
+        vals[4] = 5.0; vals[5] = 7.0; vals[6] = 1.0; vals[7] = 1.0;
+        vals[8] = 3.0; vals[9] = 4.0; vals[10] = 1.0; vals[11] = 1.0;
+        vals[12] = 4.0; vals[13] = 5.0;
+        // clang-format on
```

Comment:
Maybe you can write what the matrix is, in the comments ? For example something like below:

```
2 0 0 0
1 2 0 0
4 0 5 0
0 0 7 1

1 0 0 0
3 4 0 0
1 0 1 0
0 0 4 5
```

Reply:
Done